lib/traits/lock_widget.dart

```diff
@@ -54,6 +71,6 @@ class LockWidget extends StatelessWidget {
 
   void _lockTap(LockProvider provider) {
     bool setLock = !provider.isLocked;
-    provider.setLockUnlockAction(provider.deviceDetail.id, setLock);
+    provider.setLockUnlockAction(provider.deviceDetail?.id ?? '', setLock);
```

Comment:
Can the action be called with empty parameter. This doesn't seem right setup.

Reply:
It appears to create some sort of closure because I thought it would be odd too but verified with the debugger that the device id makes it into the repository call.